File: screen-switcher/src/main/java/com/jaynewstrom/screenswitcher/Preconditions.java
package com.jaynewstrom.screenswitcher;

import javax.annotation.Nullable;

final class Preconditions {

    static <T> T checkNotNull(@Nullable T t, String message) {
        if (t == null) {
            throw new NullPointerException(message);
        }
        return t;
    }

    static void checkArgument(boolean expectedArgument, String message) {
        if (!expectedArgument) {
            throw new IllegalArgumentException(message);
        }
    }

    static void checkState(boolean expectedState, String message) {
        if (!expectedState) {
            throw new IllegalStateException(message);
        }
    }
}

File: screen-switcher/src/main/java/com/jaynewstrom/screenswitcher/Preconditions.kt
package com.jaynewstrom.screenswitcher

internal inline fun checkArgument(expected: Boolean, crossinline messageProvider: () -> String) {
    if (!expected) {
        throw IllegalArgumentException(messageProvider())
    }
}

internal inline fun checkState(expected: Boolean, crossinline messageProvider: () -> String) {
    if (!expected) {
        throw IllegalStateException(messageProvider())
    }
}
